.github/workflows-source/schedule-daily.yml

```diff
@@ -230,3 +230,82 @@ jobs:
+        uses: ./.github/actions/bazel
+        with:
+          run: |
+            rootdir=$(mktemp -d)
+            pushd "$rootdir"
+
+            # Create dummy CUP and dummy local store
+            # Keep it base64 encoded in variables to avoid unsafe bytes
+            RECOVERY_CUP_CONTENT_B64=$(head /dev/urandom | head -c 100 | base64 | tr -d '\n')
+            RECOVERY_STORE_CONTENT1_B64=$(head /dev/urandom | head -c 100 | base64 | tr -d '\n')
+            RECOVERY_STORE_CONTENT2_B64=$(head /dev/urandom | head -c 100 | base64 | tr -d '\n')
+
+            # Pack the dummy CUP and local store into an archive
+            mkdir -p guest-os/recovery-artifacts && cd "$_"
+            echo -n "${RECOVERY_CUP_CONTENT_B64}" | base64 -d > cup.proto
+            mkdir ic_registry_local_store
+            cd ic_registry_local_store
+            mkdir -p 0001020304/05/06 && echo -n "${RECOVERY_STORE_CONTENT1_B64}" | base64 -d > "$_"/07.pb
+            mkdir -p 08090a0b0c/0d/0e && echo -n "${RECOVERY_STORE_CONTENT2_B64}" | base64 -d > "$_"/0f.pb
+            tar zcf ../ic_registry_local_store.tar.zst *
+            cd ..
+            tar zcf recovery.tar.zst cup.proto ic_registry_local_store.tar.zst
+            rm -rf cup.proto ic_registry_local_store ic_registry_local_store.tar.zst
+
+            # Upload the archive
+            popd
+            IS_GUESTOS_RECOVERY_ENGINE_SMOKE_TEST=1 bazel run //:artifact-uploader -- "$rootdir"
+
+            # Get version and checksum
+            bazel build //bazel:version.txt
+            RECOVERY_VERSION="$(cat bazel-bin/bazel/version.txt)"
+            RECOVERY_HASH="$(sha256sum $rootdir/guest-os/recovery-artifacts/recovery.tar.zst | cut -d' ' -f1)"
+
+            # Set environment variables for the next steps
+            echo "RECOVERY_CUP_CONTENT_B64=${RECOVERY_CUP_CONTENT_B64}" >> "$GITHUB_ENV"
+            echo "RECOVERY_STORE_CONTENT1_B64=${RECOVERY_STORE_CONTENT1_B64}" >> "$GITHUB_ENV"
+            echo "RECOVERY_STORE_CONTENT2_B64=${RECOVERY_STORE_CONTENT2_B64}" >> "$GITHUB_ENV"
+            echo "RECOVERY_VERSION=${RECOVERY_VERSION}" >> "$GITHUB_ENV"
+            echo "RECOVERY_HASH=${RECOVERY_HASH}" >> "$GITHUB_ENV"
+
+      - name: Adapt guestos-recovery-engine script
+        run: |
+          sed -i "s/readonly VERSION=\"\"/readonly VERSION=\"${RECOVERY_VERSION}\"/" ic-os/components/misc/guestos-recovery/guestos-recovery-engine/guestos-recovery-engine.sh
+          sed -i "s/readonly EXPECTED_RECOVERY_HASH=\"\"/readonly EXPECTED_RECOVERY_HASH=\"${RECOVERY_HASH}\"/" ic-os/components/misc/guestos-recovery/guestos-recovery-engine/guestos-recovery-engine.sh
+
+      - name: Run test
+        uses: ./.github/actions/bazel/
+        with:
+          run: |
+            bazel test --test_tag_filters=system_test_large --test_env=RECOVERY_CUP_CONTENT_B64 --test_env=RECOVERY_STORE_CONTENT1_B64 --test_env=RECOVERY_STORE_CONTENT2_B64 //rs/tests/consensus:guestos_recovery_engine_smoke_test
```

Comment:
I think you can safely drop the `test_tag_filters` since you're passing the target explicitly with `//rs/tests/etc`.